Detekt - Resolve/Suppress All Baseline Warnings - Long methods warnings_ MediaPickerFragment

Hoossayn opened the pull request:
Parent: https://github.com/wordpress-mobile/WordPress-Android/issues/17010

This PR resolves/suppresses all complexity related LongMethod warnings for the MediaPickerFragment class (see docs [here](https://detekt.dev/docs/rules/complexity#longmethod)):

`1` x LongMethod  and `1` x ComplexMethod (Resolve: 9bbd91e8b7668915952ba5618c94bbdef1cd7f55 + 37c9fa5279d8d8560db8d77d8f958c2fb6dbdc6c + eed436edc9dc09be03b3551593ed813184a4c4eb)

-----

To test:

- There is nothing much to test here.

- Verifying that all the CI checks are successful should be enough (especially the detekt check).

- However, if you really want to be thorough, you could smoke test the WordPress and/or Jetpack apps to verify that everything works as expected on every screen that relates to these changes. Specifically, you could follow the below steps:
   
    - Go to the menu tab
    - click on media option 
    - click on the upload media button or the (+) button at the top right corner 
    - select choose from device 
    - select an image

## Regression Notes
1. Potential unintended areas of impact
     can't think of any 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)
     N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Hoossayn committed:
Refactor:Extract set up RecyclerView method for media picker fragment

Hoossayn committed:
Refactor:Extract navigate Event method for media picker fragment

Hoossayn committed:
Refactor: Remove MediaPickerFragment from baseline.xml

Hoossayn commented:
👋🏾 @ParaskP7 here we go again 😎

ParaskP7 commented on WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaPickerFragment.kt:
```diff
@@ -28,6 +28,7 @@ import androidx.recyclerview.widget.GridLayoutManager.SpanSizeLookup
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 import com.google.android.material.snackbar.Snackbar
 import org.wordpress.android.R
+import org.wordpress.android.R.string
```
Minor (🔍): This import is not needed and was probably added while you extracted the `navigateEvent(...)` function. Consider removing that and then fixing the code on line `294` with `getString(R.string.cancel)`.

ParaskP7 replied:
Fixed in: 38d215f55bbb5ddbcff4bd105d5eb90c3761e5af

FYI: @Hoossayn you can follow the same pattern and reply like that here, this helps a lot. Then, after me confirming that this commit actually fixes my comment, I can resolve this conversation.

PS: You commit message(s) are still not adhering to the space (` `) standard that I recommended above:
- Instead of: `Refactor:Remove unnecessary explicit string import`
- Do this: `Refactor: Remove unnecessary explicit string import`

Hoossayn replied:
Noted, thought you meant to stick to one

ParaskP7 commented on config/detekt/baseline.xml:
```diff
@@ -72,7 +72,6 @@
     <ID>LongMethod:BarChartViewHolder.kt$BarChartViewHolder$private fun BarChart.draw( item: BarChartItem, labelStart: TextView, labelEnd: TextView ): BarCount</ID>
     <ID>LongMethod:ClicksUseCase.kt$ClicksUseCase$override fun buildUiModel(domainModel: ClicksModel, uiState: SelectedClicksGroup): List&lt;BlockListItem></ID>
     <ID>LongMethod:HomepageSettingsDialog.kt$HomepageSettingsDialog$override fun onCreateDialog(savedInstanceState: Bundle?): Dialog</ID>
-    <ID>LongMethod:MediaPickerFragment.kt$MediaPickerFragment$override fun onViewCreated(view: View, savedInstanceState: Bundle?)</ID>
```
Suggestion (💡): Now that you refactored this `onViewCreated(...)` method and correctly removed this `LongMethod` baseline warning, I noticed that you could also remove the `ComplexMethod` related such violation, see line `21` and `ComplexMethod:MediaPickerFragment.kt`. As such, with this PR, you are effectively fixing two baseline warnings at once, smooth! 🎉

ParaskP7 replied:
Fixed in: d88041f404d4d6b60529187876d16b1a50b17929

ParaskP7 reviewed (commented):
👋 @Hoossayn !

Thank you so much for this PR. One contribution a day keeps the doctor away, right, or was that for apples?! 🎉

I reviewed and tested this PR as per the description, along with triggering CI on it via a [draft](https://github.com/wordpress-mobile/WordPress-Android/pull/17378) PR, which I just pushed on the main repo.

-----

As promised, and per your contribution, I'll keep guiding you with additional notes so that you could demonstrate your best work. So, here goes once more:

- PR Description Related:
  - Main
    - You wrote: `This PR resolved/suppress all complexity related LongMethod warnings for the WordPress module (see docs):`
    - I recommend: `This PR resolves/suppresses all complexity related LongMethod warnings for the MediaPickerFragment class (see docs):`
    - Did you note the differences? It is important to have a description that matches your work and that what you are writing is grammatically correct.
  - Sections
    - Adding `-----` will split the sections with a horizontal line, maybe you could have utilized this markdown and split the `To test:` section.
  - Testing Instructions
    - Who wrote: `However, if you really want to be thorough, you could smoke test the WordPress and/or Jetpack apps to verify that everything works as expected on every screen that relates to these changes.`
    - I recommend: `However, if you really want to be thorough, you could smoke test the WordPress and/or Jetpack apps to verify that everything works as expected on every screen that relates to these changes. Specifically, you could follow the below steps:`
    - Then, you could remove the `STEPS` and also have the `Go to the menu tab` and other related instructions on a second level indentation (just like I do with my suggestions to you here).
- Commit Related:
  - Branch Name:
    - Your naming: `17010_Detekt_Resolve_Suppress_All_Baseline_Warnings_long_method_MediaPickerFragment`
    - I recommend: `issue/17010-media-picker-fragment-detekt-baseline-warnings`
    - Did you note the differences? For example, you will notice the `issue` folder level, then followed by the issue id, all words are lowercases and split with a dash (`-`) character instead of an underscore (`_`) character. FYI: This is the patter all our engineers are using, thus it would be great if you did the same, for consistency purposes.
  - I noticed that on some of your commit messages you are adding the extra space (` `) between the action and the title, but on other you don't, for example see: [Refactor:Extract set up RecyclerView method for media picker fragment](https://github.com/wordpress-mobile/WordPress-Android/pull/17376/commits/9bbd91e8b7668915952ba5618c94bbdef1cd7f55). I would be great if you could notice that, update the wording editing the commit message before pushing and opening the PR.

-----

All the above are tips for next time so don't be too alarmed, that is, you don't need to be doing anything on them for this PR, you could just update the PR description if you like. 

However, I have a couple of additional minor (🔍) and suggestions (💡) comments for you. Please take a look at them and let me known what you think. I'll then approve this PR straight away and merge this to `trunk`. 💯

Hoossayn committed:
Refactor:Remove unnecessary explicit string import

Hoossayn committed:
Refactor:Remove ComplexMethod:MediaPickerFragment.kt... from baseline.xml

Hoossayn commented:
> 👋 @Hoossayn !
> 
> Thank you so much for this PR. One contribution a day keeps the doctor away, right, or was that for apples?! 🎉
> 
> I reviewed and tested this PR as per the description, along with triggering CI on it via a [draft](https://github.com/wordpress-mobile/WordPress-Android/pull/17378) PR, which I just pushed on the main repo.
> 
> As promised, and per your contribution, I'll keep guiding you with additional notes so that you could demonstrate your best work. So, here goes once more:
> 
> * PR Description Related:
>   
>   * Main
>     
>     * You wrote: `This PR resolved/suppress all complexity related LongMethod warnings for the WordPress module (see docs):`
>     * I recommend: `This PR resolves/suppresses all complexity related LongMethod warnings for the MediaPickerFragment class (see docs):`
>     * Did you note the differences? It is important to have a description that matches your work and that what you are writing is grammatically correct.
>   * Sections
>     
>     * Adding `-----` will split the sections with a horizontal line, maybe you could have utilized this markdown and split the `To test:` section.
>   * Testing Instructions
>     
>     * Who wrote: `However, if you really want to be thorough, you could smoke test the WordPress and/or Jetpack apps to verify that everything works as expected on every screen that relates to these changes.`
>     * I recommend: `However, if you really want to be thorough, you could smoke test the WordPress and/or Jetpack apps to verify that everything works as expected on every screen that relates to these changes. Specifically, you could follow the below steps:`
>     * Then, you could remove the `STEPS` and also have the `Go to the menu tab` and other related instructions on a second level indentation (just like I do with my suggestions to you here).
> * Commit Related:
>   
>   * Branch Name:
>     
>     * Your naming: `17010_Detekt_Resolve_Suppress_All_Baseline_Warnings_long_method_MediaPickerFragment`
>     * I recommend: `issue/17010-media-picker-fragment-detekt-baseline-warnings`
>     * Did you note the differences? For example, you will notice the `issue` folder level, then followed by the issue id, all words are lowercases and split with a dash (`-`) character instead of an underscore (`_`) character. FYI: This is the patter all our engineers are using, thus it would be great if you did the same, for consistency purposes.
>   * I noticed that on some of your commit messages you are adding the extra space (` `) between the action and the title, but on other you don't, for example see: [Refactor:Extract set up RecyclerView method for media picker fragment](https://github.com/wordpress-mobile/WordPress-Android/pull/17376/commits/9bbd91e8b7668915952ba5618c94bbdef1cd7f55). I would be great if you could notice that, update the wording editing the commit message before pushing and opening the PR.
> 
> All the above are tips for next time so don't be too alarmed, that is, you don't need to be doing anything on them for this PR, you could just update the PR description if you like.
> 
> However, I have a couple of additional minor (🔍) and suggestions (💡) comments for you. Please take a look at them and let me known what you think. I'll then approve this PR straight away and merge this to `trunk`. 💯

👋🏾 @ParaskP7, one commit a day keeps the bugs away 🥳 , followed your suggestions and made changes to the code/doc 🙌🏾

ParaskP7 reviewed (approved):
👋 @Hoossayn !

Thank you so much for applying my suggestions! 🙇 💯 🥇

I did another review on your changes and everything looks good to me. As soon as CI goes ✅ I'll merge this to `trunk`.

-----

Just a quick FYI:
- Yours: `This PR resolved/suppress all complexity related LongMethod warnings for the MediaPickerFragment class (see docs here):`
- Mine: `This PR resolves/suppresses all complexity related LongMethod warnings for the MediaPickerFragment class (see docs here):`
- Find the diff! 😅

Hoossayn commented:
> * This PR resolves/suppresses

@ParaskP7  shame on me 😭😂
